src/Plugin/Action/ApprovePendingOgMembership.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\og\Plugin\Action;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Action\ActionBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\OgAccessInterface;
+use Drupal\og\OgMembershipInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Approves a pending group membership.
+ *
+ * @Action(
+ *   id = "og_membership_approve_pending_action",
+ *   label = @Translation("Approve the pending membership(s)"),
+ *   type = "og_membership"
+ * )
+ */
+class ApprovePendingOgMembership extends ActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The OG access service.
+   *
+   * @var \Drupal\og\OgAccessInterface
+   */
+  protected $ogAccess;
+
+  /**
+   * Constructs an ApprovePendingOgMembership object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\og\OgAccessInterface $og_access
+   *   The OG access service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, OgAccessInterface $og_access) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->ogAccess = $og_access;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('og.access')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute(OgMembership $membership = NULL) {
+    if (!$membership) {
+      return;
+    }
+    $membership->setState(OgMembershipInterface::STATE_ACTIVE)->save();
```

Comment:
This is already accounted for, the access check will not allow the action to be executed if it is already active.